core/src/main/java/io/questdb/cairo/O3PartitionJob.java

```diff
@@ -52,6 +60,89 @@ public O3PartitionJob(MessageBus messageBus) {
         super(messageBus.getO3PartitionQueue(), messageBus.getO3PartitionSubSeq());
     }
 
+    public static void processParquetPartition(
+            Path pathToTable,
+            int partitionBy,
+            ReadOnlyObjList<? extends MemoryCR> oooColumns,
+            long srcOooLo,
+            long srcOooHi,
+            long partitionTimestamp,
+            long sortedTimestampsAddr,
+            TableWriter tableWriter
+    ) {
+
+        // Number of rows to insert from the O3 segment into this partition.
+        final long srcOooBatchRowSize = srcOooHi - srcOooLo + 1;
+
+        final TableRecordMetadata tableWriterMetadata = tableWriter.getMetadata();
+        try (PartitionDecoder file = new PartitionDecoder(tableWriter.getFilesFacade())) {
+            final Path path = Path.getThreadLocal(pathToTable);
+            TableUtils.setPathForPartition(path.trimTo(pathToTable.size()), partitionBy, partitionTimestamp, -1);
+            path.put(".parquet");
+            file.of(path.$());
```

Comment:
We have to use the file size saved to `_txn` file from the last successful commit so that any failed dirty appends are not visible.